plugins/catalog-backend/src/catalog/types.ts

```diff
@@ -52,8 +52,22 @@ export type EntitiesRequest = {
   credentials: BackstageCredentials;
 };
 
+/**
+ * Encapsulates either a deserialized or serialized entities to be sent in a response.
+ * @internal
+ */
+export type EntitiesResponseItems =
```

Comment:
this type is slightly wonky in that not all endpoints are able to respond with nulls etc - but maybe it's ok, or add a template param `<TNullable extends boolean = false>` and the corresponding changes inside, or whatever ...

Reply:
yep I had a param for a while but it doesn’t help too much. In practice we still need to support null and passing through non-nullable items in some places doesn't really do much more than avoiding some type fiddling in a few places, but requires more code in other

Reply:
Eh, leaving this for a potential later internal refactor